pkg/block/fetcher.go

```diff
@@ -274,8 +291,84 @@ func (f *BaseFetcher) loadMeta(ctx context.Context, id ulid.ULID) (*metadata.Met
 	return m, nil
 }
 
+func (f *BaseFetcher) newCachedDeletionMark(m metadata.DeletionMark, now time.Time) *cachedDeletionMark {
+	ttl := f.deletionMarkPositiveCacheEntryTTL
+
+	return &cachedDeletionMark{
+		nextCheck: now.Add(ttl/2 + time.Duration(rand.Int63n(ttl.Nanoseconds()))),
```

Comment:
So the effective one is between ttl/2 and ttl*1.5. Is it what we want right? I would have expected `deletionMarkPositiveCacheEntryTTL` to be the upper bound, not the average TTL, but I don't have strong opinions as far as we document it.

Reply:
I mostly wanted to avoid doing all rechecks at the same time, but to spread them a little. You're right that TTL could be made upper-bound instead, and then we can compute effective TTL for example as random between [TTL/2, TTL]. I also don't have strong opinions on this. I think the naming is bit confusing here.

Reply:
I think could be made a bit more clear if we would have two hardcoded values: min TTL and max TTL. Then the actual TTL is a random between min and max.